### PR TITLE
cleanups found with phpstan

### PR DESCRIPTION
**Pull request by dbu**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | mentioned in #215
| Documentation   | -
| License         | MIT

#### What's in this PR?

Adjust to new phpstan version, cleanup some phpdoc

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,8 +13,9 @@
     "require": {
         "php": "^7.1 || ^8.0",
         "php-http/httplug": "^2.0",
-        "php-http/message-factory": "^1.0",
         "php-http/message": "^1.6",
+        "php-http/message-factory": "^1.0",
+        "phpstan/phpstan": "^1.2",
```
Comment (Nyholm) [inline composer.json:18]: Oops :)

File: src/Plugin/RetryPlugin.php [hunk 2]
```diff
@@ -140,6 +141,7 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
             }
 
             $time = call_user_func($this->exceptionDelay, $request, $exception, $this->retryStorage[$chainIdentifier]);
+            \assert(is_int($time));
```
Comment (Nyholm) [inline src/Plugin/RetryPlugin.php:144]: Will this benefit the user?
If it is just to convince phpstan we can add comments.

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you. 

It looks great. Just remove phpstan from composer.json and I’ll be happy.

File: src/Plugin/RetryPlugin.php (hunk 2 above)
Comment (GrahamCampbell) [reply src/Plugin/RetryPlugin.php:144]: Better to add a `/**@var int */` or whatever. Agreed.

File: composer.json (hunk 1 above)
Comment (dbu) [reply composer.json:18]: oh, oops indeed. will remove it again.

File: src/Plugin/RetryPlugin.php (hunk 2 above)
Comment (dbu) [reply src/Plugin/RetryPlugin.php:144]: this way the user gets a warning if the user func returns the wrong thing.

but i can also put the annotation.

**Pull request merged**